### PR TITLE
Address Coverity 1297186,187,188, 190

### PR DESCRIPTION
**Pull request by FSCyborg**
These Coverity issues are all caused by the same code being copy and pasted in three different places.

There does not seem to be actual danger of the negative array access, but the changes that "fix" it are also slight optimizations (placing a few variable assignments outside loops so that it only has to call the function once).

In short, I am basically just moving code around to avoid duplicate checks that coverity is suspicious of, and in other places checking for the negative array index explicitly.

### CODE REVIEW

**Comment by FSCyborg** [comment]
Please hold while I fix the tiny thing I did wrong.

File: code/network/multi_voice.cpp [hunk 1]
```diff
@@ -1774,12 +1777,30 @@ void multi_voice_client_send_pending()
 
 	// stream all buffered up packets
 	str = &Multi_voice_stream[0];
+
+	// get the current messaging mode
+	msg_mode = (ubyte)Multi_voice_send_mode;
+
+	// get the specific target if we're in MSG_TARGET mode
+	target_index = -1;
+	target_signature = 0; // Cyborg17 - 0 is the invalid value for net_signature
```
Comment (asarium) [inline code/network/multi_voice.cpp]: Same as before.

File: code/network/multi_voice.cpp [hunk 2]
```diff
@@ -1824,9 +1832,10 @@ void multi_voice_client_send_pending()
 
 		// add the routing data and any necessary targeting information
 		ADD_DATA(msg_mode);
-		if(msg_mode == MULTI_MSG_TARGET){
-			Assert(Game_mode & GM_IN_MISSION);
-			ADD_USHORT(Objects[Net_players[target_index].m_player->objnum].net_signature);
+
+		// Cyborg17 - add the target signature only if it's been set, which only happens in MSG_TARGET mode
+		if (target_signature){
```
Comment (asarium) [inline code/network/multi_voice.cpp]: Same as before.

**Comment by FSCyborg** [comment]
> Can you change them from `while(0)` to `while(false)`?

I went ahead and looked for all instances in the code that used while(0) and switched them over so that those won't give us a problem in the future either.  I put them in a separate commit so that it can be reverted separately if needed, or switched to another PR fairly easily, if requested.

If you guys would like me to separate those changes, or limit them to the macros involved, please let me know.

**Comment by Goober5000** [comment]
I think that's fine.  I would have put them in a different PR, but what actually matters is the fact that they're in a separate commit.  If one ever needs to track down a bug using `git bisect`, having different changes in different commits is ideal.

That being said, you should roll back the changes in `lib/libpng/pngdebug.h` and `lib/libsdl/include/SDL_assert.h`.  Those are third-party libraries so they should be left in their original state when possible.  Since all you changed were two comments, rolling them back should do no harm.

**Comment by FSCyborg** [comment]
Should now be fixed.

**Pull request merged**